src/index.js

```diff
@@ -1,36 +1,57 @@
-'use strict'
-
-const core = require('@actions/core')
-const differenceInDays = require('date-fns/differenceInDays')
-const { logInfo, logError } = require('./log')
-const { getLastReleaseDate } = require('./getLastReleaseDate')
-const { getLastRepoUpdate } = require('./getLastRepoUpdate')
-const { createIssue } = require('./createIssue')
+const core = require('@actions/core');
+const { logInfo } = require('./log');
+const { getLatestRelease, getUnreleasedCommits } = require('./release');
+const { createIssue, getLastOpenPendingIssue, updateLastOpenPendingIssue } = require('./utils');
 
 async function run() {
   try {
-    const daysToStaleRelease = core.getInput('days-to-stale-release')
-
-    const { lastReleaseDate, npmError, npmPckNotFound } = await getLastReleaseDate()
-
-    if (npmPckNotFound) return logError('npm package not found')
-
-    if (npmError) return logError(npmError)
-
-    const lastRepoUpdate = await getLastRepoUpdate()
-
-    const daysSinceRelease = differenceInDays(lastRepoUpdate, lastReleaseDate)
-
-    logInfo(`${daysSinceRelease} Days since last release`)
-
-    const shouldCreateIssue = daysToStaleRelease < daysSinceRelease
+    const token = core.getInput('github-token', { required: true });
+    const daysInput = core.getInput('days-to-ignore');
+    const daysToIgnore = Number(daysInput);
```

Comment:
avoid declaring the daysInput variable which is only used once. also, name the variable after the input name, which considering the suggestion to change it, should be staleDays

```suggestion
    const staleDays = Number(core.getInput('stale-days'));
```